EU4toV2/Source/V2World/Province/Province.cpp

```diff
@@ -50,7 +54,7 @@ V2::Province::Province(std::string _filename,
 		}
 	}
 
-	if (details.terrain.empty())
+	if (details.terrain.empty() && !theConfiguration.isHpmEnabled())
```

Comment:
Wouldn't it be better to update the terrain mapper to pull from a HPM file?

Reply:
No. This piece is reading data from configurables/terrain_data.txt assigning it to provinces for which ProvinceDetailsMapper couldn't parse terrain from /histoy/provinces (because it's not defined there). Its purpose is to override map terrain data which is not needed for HPM.

Reply:
Perhaps a comment explaining why would be good, then. Is it that HPM provinces always have the data defined?

Reply:
It does, but so too does Vanilla. It's the terrain.bmp file in map folder.

Reply:
If you check the code, terrain.bmp isn't checked at all. The converter looks for ```terrain = whatever``` in the province history files instead. Thus the backup here.

Reply:
Not the Converter, the game checks terrain.bmp to assign terrain for province. Terrain entry in province history file only serves the purpose of overriding map data. I assume they introduced it to facilitate the process of adding new provinces, instead of editing the bmp file, just define it in history file.